### PR TITLE
Shadows - fallback if depth extension is not supported

### PR DESCRIPTION
**Pull request by lilleyse**
For #2594 

As a fallback, shadows can pack their depth into a RGBA texture like in `GlobeDepth.js`. I tested this in IE and it works.

In the process of supporting this, I updated `Model.js` to cache the shadow-cast program. The model's draw commands store the shadow-cast program, which `Scene.executeShadowMapCommands` uses instead of the model's normal program. I found this easier than maintaining two command lists for render-phase/shadow-phase.

### CODE REVIEW

**Comment by pjcozzi** [comment]
All looks good to me.  @bagnell please merge when ready.

File: Source/Scene/ShadowMap.js [hunk 1]
```diff
@@ -150,69 +164,102 @@ define([
...
-    function updateFramebuffers(shadowMap, context) {
-        var depthStencilTexture = shadowMap._depthStencilTexture;
-        var textureChanged = !defined(depthStencilTexture) || (depthStencilTexture.width !== shadowMap._size);
-        if (!defined(shadowMap.framebuffer) || textureChanged) {
-            destroyTextures(shadowMap);
-            destroyFramebuffers(shadowMap);
-            createTextures(shadowMap, context);
-            createFramebuffers(shadowMap, context);
+    function updateFramebuffer(shadowMap, context) {
+        if (!defined(shadowMap.framebuffer) || (shadowMap._shadowMapTexture.width !== shadowMap._size)) {
```
Comment (bagnell) [inline Source/Scene/ShadowMap.js]: Is this supposed to be `shadowMap._framebuffer`? Right now, the framebuffer is being created and destroyed every frame.

**Comment by bagnell** [comment]
Just the one comment; otherwise, this looks good.

File: Source/Scene/ShadowMap.js (hunk 1 above)
Comment (lilleyse) [inline Source/Scene/ShadowMap.js]: Yes.. thanks for finding that.

**Comment by lilleyse** [comment]
Updated.

**Pull request merged**